source/AI.cpp

```diff
@@ -4126,6 +4126,52 @@ void AI::MovePlayer(Ship &ship, Command &activeCommands)
 			}
 		}
 	}
+	else if(activeCommands.Has(Command::SELECT_PLANET) && !ship.IsEnteringHyperspace())
+	{
+		// Cycle through possible landing sites:
+		// Track all possible landable objects in the current system.
+		auto landables = vector<const StellarObject *>{};
+		for(const StellarObject &object : ship.GetSystem()->Objects())
+		{
+			if(object.HasSprite() && object.HasValidPlanet() && object.GetPlanet()->IsAccessible(&ship))
+				landables.emplace_back(&object);
+		}
+
+		// If the player has selected a planet in this system, cycle from there.
+		const StellarObject *target = ship.GetTargetStellar();
+		auto landIt = find(landables.cbegin(), landables.cend(), target);
+		if(landIt == landables.cend())
+			landIt = landables.cbegin(); // No, so pick first planet.
```

Comment:
This comment could be a bit clearer. Also, comments should go on a separate line.